shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -735,7 +735,7 @@ private PerImageReader getActiveReader() {
           // Create a new ImageReader and add it to the queue.
           return getOrCreatePerImageReader(createImageReader());
         }
-        return imageReaderQueue.peekLast();
+        return imageReaderQueue.get(imageReaderQueue.size() - 1);
```

Comment:
Need to check length here, the peek methods return null if the size isn't enough.

Reply:
Not that it much matters, but fun drive by comment to say you could use an `ArrayDeque` here instead if you wanted to keep the same convenience right? (they both implement the double ended queue interface)

Reply:
That's a good idea, I will try that.